app/Console/Commands/AdPayGetPayments.php

```diff
@@ -35,20 +38,39 @@ public function handle(AdPay $adPay): void
     {
         $calculations = collect($adPay->getPayments(now()->getTimestamp()));
 
-        $this->info('Found '.count($calculations).' calculations.');
+        Log::info('Found '.count($calculations).' calculations.');
 
         $idList = $calculations->map(function (array $amount) {
             return hex2bin($amount['event_id']);
         });
 
         $entries = EventLog::whereIn('event_id', $idList)
-            ->get()
-            ->each(function (EventLog $entry) use ($calculations) {
-                $calculation = $calculations->firstWhere('event_id', $entry->event_id);
+            ->whereNull('event_value')
+            ->get();
+
+        Log::info('Found '.count($entries).' entries that require values.');
 
-                $entry->update(['event_value' => $calculation['amount']]);
+        $ledgerEntries = $entries->groupBy(function (EventLog $entry) {
+            return $entry->advertiser()->id;
+        })->map(function (Collection $collection, int $userId) use ($calculations) {
+            $collection->each(function (EventLog $entry) use ($calculations) {
+                $calculation = $calculations->firstWhere('event_id', $entry->event_id);
+                $entry->event_value = $calculation['amount'];
+                $entry->save();
             });
 
-        $this->info('Updated '.count($entries).' entries.');
+            $totalEventValue = $collection->sum('event_value');
+
+            $userLedgerEntry = new UserLedgerEntry();
+            $userLedgerEntry->user_id = $userId;
+            $userLedgerEntry->amount = -$totalEventValue;
+            $userLedgerEntry->status = UserLedgerEntry::STATUS_ACCEPTED;
+            $userLedgerEntry->type = UserLedgerEntry::TYPE_AD_EXPENDITURE;
+            $userLedgerEntry->save();
+
+            return $userLedgerEntry;
```

Comment:
We should use `transaction` to ensure consistency ;/

Reply:
done

Reply:
done